### PR TITLE
MINOR: Add INFO logging if ZK config is not specified

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
ping @enothereska @ijuma for reviews, this is suggested by @norwood when debugging an issue with Kafka Streams.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java [hunk 1]
```diff
@@ -124,6 +124,8 @@ public void configure(Map<String, ?> configs) {
             internalTopicManager = new InternalTopicManager(
                     (String) configs.get(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG),
                     configs.containsKey(StreamsConfig.REPLICATION_FACTOR_CONFIG) ? (Integer) configs.get(StreamsConfig.REPLICATION_FACTOR_CONFIG) : 1);
+        } else {
+            log.info("Config '{}' isn't supplied and hence no internal topics will be created.", StreamsConfig.ZOOKEEPER_CONNECT_CONFIG);
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java:128]: Is there a reason why we don't fail during config validation?

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java:128]: We are not logically defining ZOOKEEPER_CONNECT_CONFIG a required config in StreamsConfig, which we debated whether or not we need to. The reasons we decided to not make a required config is that it is actually not required for all applications: think of a piping application.

Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java:128]: Thanks for the explanation.

**Comment by ijuma** [comment]
LGTM

**Comment by guozhangwang** [comment]
Unit test passed locally. Merged to trunk.

**Pull request closed without merging**